FIX raise an error if user defined categories contain duplicate values (#27328)
scikit-learn/scikit-learn: doc/whats_new/v1.4.rst

- |Enhancement| :class:`preprocessing.TargetEncoder` now supports `target_type`
  'multiclass'. :pr:`26674` by :user:`Lucy Liu <lucyleeow>`.

- |Fix| :class:`preprocessing.OneHotEncoder` and :class:`preprocessing.OrdinalEncoder`

>   raises an error if a user provides duplicated categories.

Reviewer: ```suggestion
  raise an exception if the user provided categories contain duplicates.
```

>   will raise an error if user provided categories contain duplicate elements.

Reviewer 2: ```suggestion
 raises an error if a user provides duplicated categories.
```
